libvast/vast/expression.hpp

```diff
@@ -277,6 +277,45 @@ auto inspect(Inspector&f, expression& x) {
   return f(caf::meta::type_name("expression"), x.get_data());
 }
 
+template <class F>
+struct expression_visitor {
```

Comment:
```suggestion
struct predicate_transformer {
```

Since the scope of this visitor is a little more narrow, this name might reflect the mechanics a bit better.